OpenRA.Mods.Common/Scripting/Properties/TransportProperties.cs

```diff
@@ -34,13 +34,15 @@ public TransportProperties(ScriptContext context, Actor self)
 		public void LoadPassenger(Actor a) { cargo.Load(Self, a); }
 
 		[Desc("Remove the first actor from the transport.  This actor is not added to the world.")]
-		public Actor UnloadPassenger() { return cargo.Unload(Self); }
+		public Actor UnloadPassenger() { return cargo.UnloadLastEntered(Self); }
 
 		[ScriptActorPropertyActivity]
 		[Desc("Command transport to unload passengers.")]
 		public void UnloadPassengers()
 		{
-			Self.QueueActivity(new UnloadCargo(Self, true));
+			var toUnload = cargo.Passengers.ToList();
+			toUnload.Reverse();
```

Comment:
Actually that is the entire point of this; to unload in whatever order you'd like.
And the order we want here is reversed.